gbpservice/tests/contrib/nfp_scripts/upgrade_nfp.sh

```diff
@@ -0,0 +1,146 @@
+    IpAddr_extractor=`neutron port-list|grep $PortId|awk '{print $11}'`
+    IpAddr_purge_last=${IpAddr_extractor::-1}
+    IpAddr=${IpAddr_purge_last//\"/}
+    echo "Collecting IpAddr : for $PortId"
+    echo $IpAddr 
+}
+
+function configure_vis_ip_addr_in_docker {
+    echo "Visibility VM IP address is: $IpAddr"
+    sed -i "s/VIS_VM_IP_ADDRESS/"$IpAddr"/" $NFPSERVICE_DIR/gbpservice/nfp/configurator/Dockerfile
+}
+
+function create_images {
+    source $DEVSTACK_DIR/openrc neutron service
+    unset OS_USER_DOMAIN_ID
+    unset OS_PROJECT_DOMAIN_ID
+
+    # prepare visibility image and upload it into glance
+    VISIBILITY_QCOW2_IMAGE=${VISIBILITY_QCOW2_IMAGE:-build}
+    VISIBILITY_QCOW2_IMAGE_NAME=visibility
+    InstanceName="VisibilityVM_instance"
+    create_port_for_vm $VISIBILITY_QCOW2_IMAGE_NAME
+
+    if [[ $VISIBILITY_QCOW2_IMAGE = build ]]; then
+       # edits the docker file to add visibility vm IP address
+       configure_vis_ip_addr_in_docker
+
+       # prepare visibility source, this is needed for diskimage build
+       cd /home/stack/
+       sudo rm -rf visibility
+       sudo git clone https://$GIT_ACCESS_USERNAME:$GIT_ACCESS_PASSWORD@github.com/oneconvergence/visibility.git -b $VISIBILITY_GIT_BRANCH
+       echo "Building Image: $VISIBILITY_QCOW2_IMAGE_NAME"
+       cd $DEST/gbp/gbpservice/tests/contrib/diskimage-create/
+       sudo python visibility_disk_image_create.py visibility_conf.json $DEVSTACK_DIR/local.conf
+       VISIBILITY_QCOW2_IMAGE=$(cat /tmp/image_path)
+    fi
+    echo "Uploading Image: $VISIBILITY_QCOW2_IMAGE_NAME"
+    glance image-create --name $VISIBILITY_QCOW2_IMAGE_NAME --disk-format qcow2 --container-format bare --visibility public --file $VISIBILITY_QCOW2_IMAGE
+    sleep 4
+    
+    if ! [[ -z $AsavQcow2Image ]]; then
+        gbp service-profile-create --servicetype FIREWALL --insertion-mode l3 --shared True --service-flavor service_vendor=asav,device_type=nova --vendor NFP asav_fw_profile
+
+        ASAV_QCOW2_IMAGE_NAME=asav
+        echo "Uploading Image: $ASAV_QCOW2_IMAGE_NAME"
+        glance image-create --name $ASAV_QCOW2_IMAGE_NAME --disk-format qcow2 --container-format bare --visibility public --file $AsavQcow2Image
+    fi
+}
+
+function nfp_configure_nova {
+    NOVA_CONF_DIR=/etc/nova
+    NOVA_CONF=$NOVA_CONF_DIR/nova.conf
+    source $DEVSTACK_DIR/inc/ini-config
+    iniset $NOVA_CONF DEFAULT instance_usage_audit "True"
+}
+
+function prepare_for_upgrade {
+    if [[ $FROM = advanced ]] && [[ $TO = enterprise ]]; then
+        source $DEST/gbp/devstack/lib/nfp
+
+        echo "Preparing image creation"
+        create_images
+
+        echo "Launching the Visibility VM"
+        launch_visibilityVM
+
+        nfp_logs_forword
+        nfp_configure_nova
+        
+    else
+        echo "Not supported."
+    fi
+}
+
+function delete_instance_and_image {
+    
+    # delete the instance
+    echo "Deleting the running '$2' instance."
+    nova delete $2
+    
```

Comment:
- Do we need to wait till the VM gets deleted before deleting image? Otherwise, image delete might fail.